### PR TITLE
fix(device): prevent exception on network failure

### PR DESCRIPTION
**Pull request by rokam**
Fixes: #290

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method for improved socket state management, enhancing message reception handling.

- **Bug Fixes**
	- Enhanced error handling for connection issues with a retry mechanism, improving reliability during connection failures.

### CODE REVIEW

File: midealocal/device.py [hunk 1]
```diff
@@ -201,41 +201,35 @@ def _authenticate_refresh_capabilities(self) -> None:
     def connect(self) -> bool:
         """Connect to device."""
         connected = False
-        for _ in range(3):
```
Comment (chemelli74) [inline midealocal/device.py:204]: I'm against an infinite loop here.

Comment (rokam) [reply midealocal/device.py:204]: The removal of the for loop is not infinite loop.

**Comment by wuwentao** [comment]
@rokam add some changes to improve it .
1. change log level from warning to debug, as all the device init connect will report a warning log after HA reboot
2. use while loop before socket is available, as `connect` only need when `self._socket is None`
3. update the `self._authenticate_refresh_capabilities()` order only when first time connect, if refresh_status and self.socket is not none, skip `self._authenticate_refresh_capabilities()`

**Comment by wuwentao** [comment]
update: add try/except to all the socket send/recv, try to catch the exception and close socket, the next loop will reconnect  when socket  is none.

**Comment by wuwentao** [comment]
mark to draft as the timeout and process is error, need to continue work on it. will finish it ASAP and mark to ready for review

**Comment by rokam** [comment]
Replaced by #296

**Pull request closed without merging**